From: Daniel Drake <dsd@gentoo.org>

https://bugs.gentoo.org/show_bug.cgi?id=249498
http://bugzilla.kernel.org/show_bug.cgi?id=12149

Revert commit 5b7dba4ff834259a5623e03a565748704a8fe449
sched_clock: prevent scd->clock from moving backwards

reverted:
--- b/kernel/sched_clock.c
+++ a/kernel/sched_clock.c
@@ -118,13 +118,13 @@
 
 	/*
 	 * scd->clock = clamp(scd->tick_gtod + delta,
+	 * 		      max(scd->tick_gtod, scd->clock),
+	 * 		      scd->tick_gtod + TICK_NSEC);
-	 *		      max(scd->tick_gtod, scd->clock),
-	 *		      max(scd->clock, scd->tick_gtod + TICK_NSEC));
 	 */
 
 	clock = scd->tick_gtod + delta;
 	min_clock = wrap_max(scd->tick_gtod, scd->clock);
+	max_clock = scd->tick_gtod + TICK_NSEC;
-	max_clock = wrap_max(scd->clock, scd->tick_gtod + TICK_NSEC);
 
 	clock = wrap_max(clock, min_clock);
 	clock = wrap_min(clock, max_clock);
